File: silkyanimation/src/main/java/com/yuyashuai/silkyanimation/BitmapDecoder.java
package com.yuyashuai.silkyanimation;

import android.graphics.Bitmap;
import android.support.annotation.Nullable;

import java.io.File;
import java.util.List;

/**
 * @author yuyashuai   2018/7/31.
 */
public interface BitmapDecoder {

    /**
     * get the bitmap surface need
     *
     * @return the last frame bitmap
     */
    @Nullable
    Bitmap get();

    /**
     * notify decode the bitmap of next frame
     */
    void notifyUpdate();

    /**
     * recycle all resources
     */
    void stopAndRecycle();

    /**
     * start decode bitmap to memory
     *
     * @param startOffset the start position of decoding
     * @param files    the resources directory
     */
    void startDecodeFilesDirectory(int startOffset, File... files);

    /**
     * start decode bitmap to memory
     *
     * @param startOffset the start position of decoding
     * @param files    the assets resources directory
     */
    void startDecodeAssetsDirectory(int startOffset, String... files);

    /**
     * start decode bitmap to memory
     *
     * @param startOffset     the start position of decoding
     * @param filePathList the absolutely path of all frame bitmaps
     */
    void startDecodeFilesPath(int startOffset, List<String> filePathList);

    /**
     * start decode bitmap to memory
     *
     * @param startOffset     the start position of decoding
     * @param filePathList the absolutely path of all frame bitmaps
     */
    void startDecodeAssetsPath(int startOffset, List<String> filePathList);

    /**
     * set number of frames resides in memory
     *
     * @param count number of frames resides in memory.
     */
    void setMemoryCacheCount(int count);

}

File: frameanimation/src/main/java/com/yuyashuai/frameanimation/io/BitmapDecoder.kt
package com.yuyashuai.frameanimation.io

import android.graphics.Bitmap
import com.yuyashuai.frameanimation.FrameAnimation

/**
 * @author yuyashuai   2019-04-24.
 * responsible for decoding picture
 */
interface BitmapDecoder {

    /**
     * decoding the bitmap from the specified path
     * don't allocate objects in this method
     * @param path the bitmap path
     * @param inBitmap the reuse bitmap
     * @return null when decode fails
     */
    fun decodeBitmap(path: FrameAnimation.PathData, inBitmap: Bitmap?): Bitmap?

    /**
     * @see decodeBitmap
     */
    fun decodeBitmap(path: FrameAnimation.PathData): Bitmap?
}